src/exts/discord_integration.js

```diff
@@ -0,0 +1,77 @@
+const { bridge } = require('../spring_api');
+const { log } = require('../spring_log');
+const { config } = require('../launcher_config');
+const DiscordRPC = require("discord-rpc");
+
+if (config.discord_rich_presence.application_id == null) {
+    log.warn("config.discord_rich_presence.application_id not defined");
+    return
+}
+
+// Application ID from https://discord.com/developers/applications/<applicationId>/
+const applicationId = config.discord_rich_presence.application_id;
+const rpc = new DiscordRPC.Client({ transport: 'ipc' });
+let loginSuccessful = false;
+
+DiscordRPC.register(applicationId);
+
+async function TryToLogin() {
+    try {
+        await rpc.login({ clientId: applicationId });
+        loginSuccessful = true;
+    } catch (error) {
+        log.warn("Discord RPC login error - Discord not running?");
+        loginSuccessful = false;
+    }
+}
+
+
+//if (!loginSuccessful) return;
+
+bridge.on('DiscordSetActivity', async command => {
+    // command
+    // https://discord.com/developers/docs/rich-presence/how-to#updating-presence-update-presence-payload-fields
+    // {
+    //     state : string - Playing, spectating, in menu, in lobby, watching replay...
+    //     details : string - Map name
```

Comment:
btw why is it details a not "mapName" ?

Reply:
I was trying to use the same names as in Discord's documentation.

Reply:
I realised I wasn't consistent in following this so I renamed it to mapName and partyId to battleId